hommod_rest/services/interpro.py

```diff
@@ -133,30 +131,28 @@ def _create_data_file(self, sequence):
         if not os.path.isdir(self.storage_dir):
             os.mkdir(self.storage_dir)
 
-        ID = idForSeq(sequence)  # unique
+        sequence_id = idForSeq(sequence)  # unique
 
-        outfilepath = os.path.join(self.storage_dir, '%s.xml.bz2' % ID)
+        outfilepath = os.path.join(self.storage_dir,
+                                   '%s.xml.bz2' % sequence_id)
         if os.path.isfile(outfilepath):
             return outfilepath
 
         # We don't want two processes to build the same interpro file
         # at the same time. So use a lock file:
-        lock = FileLock(self._interpro_lockfile_path(ID))
-
-        with lock:
+        with FileLock(self._interpro_lockfile_path(sequence_id)) as lock:
             # Wait for a place in line to start an interpro job
             while self._interpro_count_lockfiles() >= _MAX_INTERPRO_JOBS:
                 sleep(10)
 
             # Wait for the interpro server to finish:
             jobid = _interpro_submit(sequence)
-            while True:
-
+            start_time = time()
+            while (time() - start_time) < 1000:
```

Comment:
Two improvements here:

1. import the entire `time` module, since functions like `time()` and `sleep()` could easily be imported from other modules without realising it. It's better to keep the `time` namespace.
2. Instead of `1000` (which is termed a magic number, because it doesn't convey meaning), use the following: `while time.time() - start_time < time.timedelta(seconds=1000)`. This makes it clear that 1000 refers to the number of seconds.